### PR TITLE
embedded table: various improvements

### PR DESCRIPTION
**Pull request by goneri**
With this commit, it's now possible to embbed different row of the same table.

We use the DOMAIN to know the name resource to embed. This instead of the
name of the field.

### CODE REVIEW

**Comment by goneri** [comment]
I will prepare the Unit-test if you are fine with the patch.

File: eve_sqlalchemy/utils.py [hunk 1]
```diff
@@ -62,17 +62,20 @@ def sqla_object_to_dict(obj, fields):
...
-                    result[field] = sqla_object_to_dict(val, list(config.DOMAIN[field]['schema'].keys()))
-                else:
+                try:
+                    foreign_resource = config.DOMAIN[resource]['schema'][field]['data_relation']['resource']
+                    foreign_fields = list(config.DOMAIN[foreign_resource]['schema'].keys())
+                except KeyError:
                     result[field] = getattr(val, config.ID_FIELD)
+                else:
+                    # we have embedded document in schema, let's resolve it:
+                    result[field] = sqla_object_to_dict(val, foreign_fields, foreign_resource)
```
Comment (kiniou) [inline eve_sqlalchemy/utils.py]: I think this will fail since `foreign_resource` may not be set when the `KeyError` exception occurs.

**Comment by amleczko** [comment]
I need to sleep with it :)

**Comment by goneri** [comment]
I need more time to finish this patch. i will reopen a PR later.

**Pull request closed without merging**